wasm-host-simulator/src/host_functions_wamr.rs

```diff
@@ -887,6 +887,162 @@ pub fn get_nft(
     dp_res.0
 }
 
+pub fn amendment_enabled(
+    env: wasm_exec_env_t,
+    amendment_ptr: *const u8,
+    amendment_len: usize,
+) -> i32 {
+    let data_provider = get_dp(env);
+    if amendment_len > MAX_WASM_PARAM_LENGTH {
+        return HostError::DataFieldTooLarge as i32;
+    }
+    // For now, return 0 (not enabled) as a stub implementation
+    // In a real implementation, this would check against the ledger's enabled amendments
+    0
+}
+
+pub fn check_sig(
+    _env: wasm_exec_env_t,
+    message_ptr: *const u8,
+    message_len: usize,
+    signature_ptr: *const u8,
+    signature_len: usize,
+    pubkey_ptr: *const u8,
+    pubkey_len: usize,
+) -> i32 {
+    if message_len > MAX_WASM_PARAM_LENGTH
+        || signature_len > MAX_WASM_PARAM_LENGTH
+        || pubkey_len > MAX_WASM_PARAM_LENGTH
+    {
+        return HostError::DataFieldTooLarge as i32;
+    }
+    // For now, return 0 (invalid signature) as a stub implementation
+    // In a real implementation, this would verify the signature using the public key
+    0
+}
+
+pub fn get_base_fee(env: wasm_exec_env_t) -> i32 {
+    let data_provider = get_dp(env);
+    // For now, return a default base fee of 10 drops
+    // In a real implementation, this would retrieve the base fee from the ledger
+    10
+}
+
+pub fn get_ledger_account_hash(
+    env: wasm_exec_env_t,
+    out_buf_ptr: *mut u8,
+    out_buf_cap: usize,
+) -> i32 {
+    if HASH256_LEN > out_buf_cap {
+        return HostError::BufferTooSmall as i32;
+    }
+    // For now, return a zero hash as a stub implementation
```

Comment:
Unclear if these stub implementations are sufficient, or if we want to add something to the `wasm-host-simulator` here? Currently, I lean towards "do nothing except use these stubs" but could be persuaded otherwise.